design/apidsl/security.go

```diff
@@ -0,0 +1,308 @@
+package apidsl
+
+import (
+	"github.com/goadesign/goa/design"
+	"github.com/goadesign/goa/dslengine"
+)
+
+// Security defines an authentication method required to access a goa
+// Action.  When defined on a Resource, it applies to all Actions,
+// unless overriden by individual actions.  When defined at the API
+// level, it will apply to all resources by default, following the
+// same logic.
+//
+// The method name refers to what you have passed to the different
```

Comment:
I guess that will become "The method refers..." once `method` becomes `interface{}` and the code checks whether it's a `SecurityDefinition` or the name of one. At that point the code should check that the name corresponds to an existing security definition.